src/engine/enginemaster.cpp

```diff
@@ -443,22 +546,79 @@ void EngineMaster::process(const int iBufferSize) {
-        // with the channel gains and overall master gain.
-        if (!m_pMasterTalkoverMix->toBool()) {
-            // Add Talkover to Master output
-            SampleUtil::copy4WithGain(m_pMaster,
-                    m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+        // Make master and booth mixes from the
+        // crossfader orientation buffers and talkover mix
+        SampleUtil::copy3WithGain(m_pMaster,
+            m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
+            m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
+            m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+            iBufferSize);
+
+        if (configuredTalkoverMixMode != TalkoverMixMode::DIRECT_MONITOR
+            && m_pNumMicsConfigured->get() > 0) {
+              m_pInputLatencyCompensationDelay->process(m_pMaster, iBufferSize);
+        }
+
+        if (boothEnabled) {
+            CSAMPLE boothGain = m_pBoothGain->get();
+
+            if (configuredTalkoverMixMode == TalkoverMixMode::MASTER) {
+                // Mix the talkover mix with the master output, but exclude it from
+                // the booth output.
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
+                } else {
+                    SampleUtil::copy1WithGain(m_pBooth, m_pMaster,
+                        boothGain, iBufferSize);
+                }
+
+                SampleUtil::copy2WithGain(m_pMaster,
+                    m_pMaster, 1.0,
+                    m_pTalkover, 1.0,
+                    iBufferSize);
+            } else if (configuredTalkoverMixMode == TalkoverMixMode::MASTER_AND_BOOTH) {
+                // Mix the talkover mix with the master output and include
+                // it in the booth output too.
+                SampleUtil::copy2WithGain(m_pMaster,
+                    m_pMaster, 1.0,
                     m_pTalkover, 1.0,
                     iBufferSize);
+
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
+                } else {
+                    SampleUtil::copy1WithGain(m_pBooth, m_pMaster,
+                        boothGain, iBufferSize);
+                }
+            } else if (configuredTalkoverMixMode == TalkoverMixMode::DIRECT_MONITOR) {
+                // Skip mixing the talkover mix with the master mix if using direct monitoring
+                // because it is being mixed in hardware without the latency of sending
+                // the signal in to Mixxx for processing.
+                // The talkover mix will be mixed later with recording/broadcasting signal
+                // after latency compensation.
```

Comment:
> Unfortunately it does not work, because the master buffer already contains the talkover signal.

[No it does not.](https://github.com/mixxxdj/mixxx/pull/1279/files#diff-6a1e31696d30e425b38b3ada10cf33bcR595)

> You need to delay the master signal before mixing the talkover signal in .

That is what is done.

> For example pMixerOut and pTalkover and pMasterMix,

What is pMixerOut in your examples?